catkit/hardware/SnmpUps.py

```diff
@@ -48,7 +48,11 @@ def is_power_ok(self, return_status_msg=False):
                 return result
 
         except Exception as err:
-            self.log.exception(err.message)
+            # Most, but not all, exception subclasses have a message attribute. Handle either case.
+            if hasattr(err, 'message'):
+                self.log.exception(err.message)
+            else:
+                self.log.exception(err)
```

Comment:
[revision] I might be wrong, but ``BaseException`` and thus none of the builtins have a ``message`` attribute. It has an ``args`` attribute for which ``args[0]`` is the message passed upon instantiation.

Either way, we don’t have to worry about it as ``log.exception()`` takes care of this for us.
So the simplest fix would be to just drop the ``.message``:
```python
 self.log.exception(err)
```
With also the option to roll in the error message we construct a little below this, i.e.,
```python
error_message = f"{self.config_id} failed safety test: SNMP interface request failed."
self.log.exception(error_message)  # This will auto bake the traceback into the log.
```

I can take this PR from here if you'd like?

Reply:
Hey thanks, much appreciated @jamienoss. For sure if you want to  take this from here then go for it. I did wonder about whether the right fix here was just to drop the `.message` but I figured there may have been some reason that was there before, possibly some other exception that did have that attribute? Not sure. In any case, I'm more than happy to hand this one over to you. Cheers.